hyperspy/_signals/hologram_image.py

```diff
@@ -546,6 +587,174 @@ def reconstruct_phase(self,
+        'Fringe sampling (px)': 3.7902017241882331,
+        'Carrier frequency (1 / nm)': 0.28685808994016415}
+        """
+
+        # Testing match of navigation axes of reference and self
+        # (exception: reference nav_dim=1):
+
+        # Parsing sideband position:
+        (sb_position, sb_position_temp) = _parse_sb_position(self, None, sb_position, sb, high_cf, parallel)
+
+        # Calculate carrier frequency in 1/px and fringe sampling:
+        fourier_sampling = 1./np.array(self.axes_manager.signal_shape)
+        if single_values:
+            carrier_freq_px = calculate_carrier_frequency(_first_nav_pixel_data(self),
+                                                          sb_position=_first_nav_pixel_data(sb_position),
+                                                          scale=fourier_sampling)
+        else:
+            carrier_freq_px = self.map(calculate_carrier_frequency,
+                                       sb_position=sb_position,
+                                       scale=fourier_sampling,
+                                       inplace=False,
+                                       ragged=False,
+                                       show_progressbar=show_progressbar,
+                                       parallel=parallel)
+        fringe_sampling = np.divide(1., carrier_freq_px)
+
+        ureg = UnitRegistry()
+        try:
+            units = ureg.parse_expression(str(self.axes_manager.signal_axes[0].units))
+        except UndefinedUnitError:
+            raise ValueError('Signal axes units should be defined')
+
+        # Calculate carrier frequency in 1/units and fringe spacing in units:
+        f_sampling_units = np.divide(
+            1.,
+            [a * b for a, b in
+             zip(self.axes_manager.signal_shape,
+                 (self.axes_manager.signal_axes[0].scale,
+                  self.axes_manager.signal_axes[1].scale))]
+        )
+        if single_values:
+            carrier_freq_units = calculate_carrier_frequency(_first_nav_pixel_data(self),
+                                                          sb_position=_first_nav_pixel_data(sb_position),
+                                                          scale=f_sampling_units)
+        else:
+            carrier_freq_units = self.map(calculate_carrier_frequency,
+                                       sb_position=sb_position,
+                                       scale=f_sampling_units,
+                                       inplace=False,
+                                       ragged=False,
+                                       show_progressbar=show_progressbar,
+                                       parallel=parallel)
+        fringe_spacing = np.divide(1., carrier_freq_units)
+
+        # Calculate carrier frequency in mrad:
+        try:
+            ht = self.metadata.Acquisition_instrument.TEM.beam_energy
+        except:
+            raise AttributeError("Please define the beam energy."
```

Comment:
Typo: "." at the end of the sentence.

Reply:
I guess it refers to the next sentence - fixed))